Metal type not defined on Linux

The `CAMetalLayer` type isn’t defined anywhere on Linux, making build fail with this error:
```
pathfinder/c/build/include/pathfinder/pathfinder.h:354:38: error: ‘CAMetalLayer’ was not declared in this scope
```

It would be useful to `#ifdef` out the Metal functions on unsupported OSes.

This is a blocker for #258.

--- a/c/cbindgen.toml
+++ b/c/cbindgen.toml
@@ -12,6 +12,10 @@ header = """\
 #ifdef __cplusplus
 extern \"C\" {
 #endif
+
+#ifndef __APPLE__
+typedef struct CAMetalLayerPrivate CAMetalLayer;
+#endif
 """
 trailer = """\
 #ifdef __cplusplus

